File: app/src/main/java/me/tatarka/betterfragment/sample/SimpleHostFragment.java
package me.tatarka.betterfragment.sample;

import android.os.Bundle;
import android.view.View;

import javax.inject.Inject;

import androidx.annotation.Nullable;
import me.tatarka.betterfragment.app.Fragment;
import me.tatarka.betterfragment.widget.FragmentHost;

public class SimpleHostFragment extends Fragment {

    private final Factory fragmentFactory;

    @Inject
    public SimpleHostFragment(Factory factory) {
        fragmentFactory = factory;
    }

    @Override
    public void onCreate(@Nullable Bundle savedState) {
        super.onCreate(savedState);
        setContentView(R.layout.simple_host);
        final FragmentHost host = requireViewById(R.id.host);
        requireViewById(R.id.one).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                host.setFragment(fragmentFactory.newInstance(MyFragment.class).withNumber(1));
            }
        });
        requireViewById(R.id.two).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                host.setFragment(fragmentFactory.newInstance(MyFragment.class).withNumber(2));
            }
        });
    }
}

File: app/src/main/java/me/tatarka/betterfragment/sample/SimpleHostFragment.kt
package me.tatarka.betterfragment.sample

import android.view.View
import me.tatarka.betterfragment.app.Fragment
import me.tatarka.betterfragment.sample.dagger.DaggerFragmentFactory
import me.tatarka.betterfragment.wiget.FragmentHost
import javax.inject.Inject

class SimpleHostFragment @Inject constructor(
    private val fragmentFactory: DaggerFragmentFactory
) : Fragment() {

    override fun onCreate() {
        super.onCreate()
        setContentView(R.layout.simple_host)
        val host: FragmentHost = requireViewById(R.id.host)
        requireViewById<View>(R.id.one).setOnClickListener {
            host.fragment =
                    fragmentFactory.newInstance<MyFragment>().withNumber(1)
        }
        requireViewById<View>(R.id.two).setOnClickListener {
            host.fragment =
                    fragmentFactory.newInstance<MyFragment>().withNumber(2)
        }
    }
}
